xdsl/dialects/builtin.py

```diff
@@ -334,6 +334,12 @@ def __init__(
 i1 = IntegerType(1)
 
 
+SignlessIntegerConstraint = ParamAttrConstraint(
```

Comment:
Uncertain about the name and placement of this.
Thanks to @PapyChacal who showed me a similar way to do this.

Reply:
I would also add
```
SignlessIntegerType: TypeAlias = Annotated[IntegerType, SignlessIntegerConstraint]
```
Then you can use `SignlessIntegerType` in place of `IntegerType` when you need python types instead of python objects.

Reply:
The placement and name are good I think.